### PR TITLE
Setup basic build with release and javadoc publishing

### PR DESCRIPTION
**Pull request by lauraluiz**
@schleichardt 
- Project with IT configured
- Javadoc publishing (missing Travis side)
- Basic release (missing to publish it to a remote repo)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+def configTestDirs(config: Configuration, folderName: String) = Seq(
+  javaSource in config := baseDirectory.value / folderName,
+  scalaSource in config := baseDirectory.value / folderName,
+  resourceDirectory in config := baseDirectory.value / s"$folderName/resources"
+)
+
+def configCommonTestSettings(scopes: String) = Seq(
+  testOptions += Tests.Argument(TestFrameworks.JUnit, "-v"),
+  libraryDependencies ++= Seq (
+    "org.assertj" % "assertj-core" % "3.0.0" % scopes
```
Comment (schleichardt) [inline build.sbt]: I think there is already 3.4.1 out.

Comment (lauraluiz) [inline build.sbt]: I'll check all dependencies

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ),
+  dependencyOverrides ++= Set (
+    "junit" % "junit" % "4.12" % scopes
```
Comment (schleichardt) [inline build.sbt]: can you add two dummy tests for test and it? Sometimes SBT is bitchy concerning runnning jUnit tests. I expected this as dependency: https://github.com/sbt/junit-interface

http://www.scala-sbt.org/0.13/docs/Testing.html#JUnit

**Comment by schleichardt** [comment]
👍

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+lazy val `commercetools-sunrise-email` = (project in file("."))
+  .configs(IntegrationTest)
+  .settings(commonSettings ++ commonTestSettings : _*)
+
+lazy val commonSettings = releaseSettings ++ Seq (
+  scalaVersion := "2.11.7",
+  javacOptions ++= Seq("-source", "1.8", "-target", "1.8")
+)
+
```
Comment (schleichardt) [inline build.sbt:15]: could you also create a spot for production dependencies?

File: build.sbt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  javaSource in config := baseDirectory.value / folderName,
+  scalaSource in config := baseDirectory.value / folderName,
+  resourceDirectory in config := baseDirectory.value / s"$folderName/resources"
+)
+
+def configCommonTestSettings(scopes: String) = Seq(
+  testOptions += Tests.Argument(TestFrameworks.JUnit, "-v"),
+  libraryDependencies ++= Seq (
+    "org.assertj" % "assertj-core" % "3.0.0" % scopes
+  ),
```
Comment (schleichardt) [inline build.sbt]: you could also add the dependency to http://www.icegreen.com/greenmail/

File: build.sbt (hunk 2 above)
Comment (schleichardt) [inline build.sbt]: the tests could be part of the .travis.yml build

Comment (lauraluiz) [inline build.sbt]: Funny, we have the other one in Sunrise. I'll create an issue to change it.

Comment (lauraluiz) [inline build.sbt]: Ah, I see it's a overridden dependency what we have in Sunrise. I guessed it was necessary for AsserJ, but I guess it's more Play related?

File: build.sbt (hunk 4 above)
Comment (lauraluiz) [inline build.sbt]: done

File: build.sbt (hunk 2 above)
Comment (lauraluiz) [inline build.sbt]: dummy tests done

File: build.sbt (hunk 3 above)
Comment (lauraluiz) [inline build.sbt:15]: I could, but it will be empty now. What kind of dependencies do you have in mind?

**Pull request merged**